### PR TITLE
GH-44: Fix a malloc error when providing the bakcup's path

### PR DESCRIPTION
**Pull request by didix21**

### CODE REVIEW

**Review by irikuo — APPROVED** [review]
LGTM

File: src/binding.cc [hunk 1]
```diff
@@ -16,8 +16,8 @@ namespace idevice_info_node {
     char *ToCString(Isolate* isolate, const Local<Value>& value) {
         if (!value->IsString()) return NULL;
         String::Utf8Value string(isolate, value);
-        char *str = (char *) malloc(string.length() - 1);
-        strcpy(str, *string);
+        char *str = (char *) malloc(string.length() + 1);
+        strncpy(str, *string, string.length() + 1);
```
Comment (jesusmaloQ) [inline src/binding.cc:20]: The value of str could be null if malloc fails, so this code could generate a SIGSEGV. Fix it.

**Review by jesusmaloQ — CHANGES_REQUESTED** [review]
The proposal should be rejected. Poor error handling.

File: src/binding.cc (hunk 1 above)
Comment (didix21) [reply src/binding.cc:20]: Right, I fix it. Thanks

File: src/binding.cc [hunk 2]
```diff
@@ -16,8 +16,14 @@ namespace idevice_info_node {
     char *ToCString(Isolate* isolate, const Local<Value>& value) {
         if (!value->IsString()) return NULL;
         String::Utf8Value string(isolate, value);
-        char *str = (char *) malloc(string.length() - 1);
-        strcpy(str, *string);
+        char *str = (char *) malloc(string.length() + 1);
```
Comment (jesusmaloQ) [inline src/binding.cc:19]: The argument to malloc should be `sizeof(char) * (string.length() + 1)`

**Review by jesusmaloQ — CHANGES_REQUESTED** [review]

**Pull request merged**